ee/tabby-webserver/src/service/mod.rs

```diff
@@ -60,41 +63,31 @@ impl ServerContext {
         }
     }
 
-    async fn authorize_request(&self, request: &Request<Body>) -> bool {
+    async fn authorize_request(&self, request: &Request<Body>) -> (bool, Option<String>) {
         let path = request.uri().path();
-        if path.starts_with("/v1/") || path.starts_with("/v1beta/") {
-            let token = {
-                let authorization = request
-                    .headers()
-                    .get("authorization")
-                    .map(HeaderValue::to_str)
-                    .and_then(Result::ok);
-
-                if let Some(authorization) = authorization {
-                    let split = authorization.split_once(' ');
-                    match split {
-                        // Found proper bearer
-                        Some(("Bearer", contents)) => Some(contents),
-                        _ => None,
-                    }
-                } else {
-                    None
-                }
-            };
-
-            if let Some(token) = token {
-                if self.db_conn.verify_access_token(token).await.is_err()
-                    && !self.db_conn.verify_auth_token(token).await
-                {
-                    return false;
-                }
-            } else {
-                // Admin system is initialized, but there's no valid token.
-                return false;
-            }
+        if !(path.starts_with("/v1/") || path.starts_with("/v1beta/")) {
+            return (true, None);
+        }
+        let authorization = request
+            .headers()
+            .get("authorization")
+            .map(HeaderValue::to_str)
+            .and_then(Result::ok);
+
+        let token = authorization
+            .and_then(|s| s.split_once(' '))
+            .map(|(_bearer, token)| token);
+
+        let Some(token) = token else {
+            return (false, None);
+        };
+        if self.db_conn.verify_access_token(token).await.is_ok() {
```

Comment:
it's always possible to extract email from access token, it's being filled in the `sub` field:

https://github.com/TabbyML/tabby/blob/1f51a0946a9aa9f2b3bf52931b9da5ca65225439/ee/tabby-webserver/src/schema/auth.rs#L254

Reply:
Isn't the return being `(true, None)` for unauthenticated routes still required, though?

Reply:
ah, you're right, we still to maintain them (everything other than `/v1/` `/v1beta/` don't have to do authorization through this middle ware).

Reply:
let's fix the verify_access_token and this PR is good to go

Reply:
Already did with [07a0124](https://github.com/TabbyML/tabby/pull/1341/commits/07a0124434f17b1073f63747df39b984cb48bb92)